src/Symfony/Component/Yaml/Inline.php

```diff
@@ -371,11 +371,11 @@ private static function parseSequence(string $sequence, int $flags, int &$i = 0,
                     $value = self::parseMapping($sequence, $flags, $i, $references);
                     break;
                 default:
-                    $isQuoted = \in_array($sequence[$i], ['"', "'"]);
+                    $isQuoted = $sequence[$i] === '"' || $sequence[$i] === "'";
                     $value = self::parseScalar($sequence, $flags, [',', ']'], $i, null === $tag, $references);
 
                     // the value can be an array if a reference has been resolved to an array var
-                    if (\is_string($value) && !$isQuoted && false !== strpos($value, ': ')) {
+                    if (!$isQuoted && \is_string($value) && false !== strpos($value, ': ')) {
```

Comment:
isn't `\is_string($value)` very fast thanks to being optimized at the opcode level ? I think this does not actually improve performance much here, and so changing it is not worth it (while it has a maintenance cost due to potential conflicts when merging branches)